src/supabase/queries/auth.ts

```diff
@@ -57,3 +51,45 @@ export const signOutUser = async () => {
     throw error;
   }
 };
+
+export const deleteCurrentUser = async (userId: string) => {
+  try {
+    await supabaseAdmin.auth.admin.deleteUser(userId);
+  } catch (error) {
+    throw error;
+  }
+};
+
+export const getCurrentUserInfo = async () => {
+  try {
+    const {
+      data: { user },
+    } = await supabase.auth.getUser();
+    return user;
+  } catch (error) {
+    throw error;
+  }
+};
+
+export const updateCurrUserName = async (
+  newFirstName: string,
+  newLastName: string,
+) => {
+  try {
+    await supabase.auth.updateUser({
+      data: { firstName: newFirstName, lastName: newLastName },
```

Comment:
Let's also add the ability to update middle name here since we have this as a field in our sign up flow!